### PR TITLE
Allow multiple apps with same repo

### PR DESCRIPTION
**Pull request by satr**

### CODE REVIEW

File: api/applications/models/application_registration.go [hunk 1]
```diff
@@ -61,4 +61,9 @@ type ApplicationRegistration struct {
 	//
 	// required: true
 	ConfigBranch string `json:"configBranch"`
+
+	// AcknowledgeWarnings acknowledge all warnings
+	//
+	// required: false
+	AcknowledgeWarnings bool `json:"acknowledgeWarnings,omitempty"`
```
Comment (nilsgstrabo) [inline api/applications/models/application_registration.go:68]: Add ApplicationRegistrationResponse struct and move AcknowledgeWarnings as field to this struct, together with ApplicationRegistration.

File: api/applications/models/application_registration_upsert_result.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package models
+
+// ApplicationRegistrationUpsertResult describe an application upsert operation result
+// swagger:model ApplicationRegistrationUpsertResult
+type ApplicationRegistrationUpsertResult struct {
```
Comment (nilsgstrabo) [inline api/applications/models/application_registration_upsert_result.go:5]:
```suggestion
type ApplicationRegistrationUpsertResponse struct {
```

File: api/applications/models/application_patchrequest.go [hunk 3]
```diff
@@ -34,4 +34,9 @@ type ApplicationPatchRequest struct {
 	//
 	// required: false
 	ConfigBranch *string `json:"configBranch,omitempty"`
+
+	// AcknowledgeWarnings acknowledge all warnings
+	//
+	// required: false
+	AcknowledgeWarnings bool `json:"acknowledgeWarnings,omitempty"`
```
Comment (nilsgstrabo) [inline api/applications/models/application_patchrequest.go:41]: Refactor ApplicationPatchRequest: add ApplicationPatch struct containing fields to be patched, and include ApplicationPatch as field Patch in ApplicationPatchRequest

**Review by nilsgstrabo — CHANGES_REQUESTED** [review]
see comments

File: api/applications/models/application_registration.go (hunk 1 above)
Comment (satr) [reply api/applications/models/application_registration.go:68]: I guess it expected to be `ApplicationRegistrationRequest`

Comment (satr) [reply api/applications/models/application_registration.go]: Fixed

File: api/applications/models/application_registration_upsert_result.go (hunk 2 above)
Comment (satr) [reply api/applications/models/application_registration_upsert_result.go]: Fixed

File: api/applications/models/application_registration_upsert_result.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package models
+
+// ApplicationRegistrationUpsertResult describe an application upsert operation result
+// swagger:model ApplicationRegistrationUpsertResult
```
[the remark this answers is not in the document]
Comment (satr) [reply api/applications/models/application_registration_upsert_result.go]: Fixed

File: api/applications/models/application_registration_upsert_result.go [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package models
+
+// ApplicationRegistrationUpsertResult describe an application upsert operation result
```
[the remark this answers is not in the document]
Comment (satr) [reply api/applications/models/application_registration_upsert_result.go]: Fixed

File: api/applications/models/application_patchrequest.go (hunk 3 above)
Comment (satr) [reply api/applications/models/application_patchrequest.go]: Fixed

**Review by nilsgstrabo — APPROVED** [review]

**Pull request merged**